src/types/led_location.rs

```diff
@@ -78,33 +79,22 @@ impl LedLocation {
     /// ```
     #[allow(clippy::new_ret_no_self)]
     pub fn new(row: u8, common: u8) -> Result<Self, ValidationError> {
-        if row >= ROWS_SIZE as u8 {
-            return Err(ValidationError::ValueTooLarge {
+        let row = DisplayDataAddress::try_from(row)
+            .map_err(|_| ValidationError::ValueTooLarge {
                 name: "row",
                 value: row,
                 limit: ROWS_SIZE as u8,
                 inclusive: false,
-            });
-        }
-
-        if common >= COMMONS_SIZE as u8 {
-            return Err(ValidationError::ValueTooLarge {
-                name: "common",
-                value: common,
-                limit: COMMONS_SIZE as u8,
-                inclusive: false,
-            });
-        }
+            })?;
 
-        let row = DisplayDataAddress::from_bits_truncate(row);
-        let common = DisplayData::from_bits_truncate(1 << common);
+        let common = DisplayData::try_from_common(common)?;
```

Comment:
Instead of making checks against invalid commons here I cut the code segment into a reusable method.